pyproject.toml

```diff
@@ -94,15 +94,17 @@ dependencies = [
 
 [project.optional-dependencies]
 dev = [
-    "black==22.3.0",
+    "black>=22.7.0",
     "coverage~=5.5",
     "ruff==0.0.262",
-    "isort~=5.0",
-    "mypy==0.812",
-    "mypy-zope==0.3.0",
+    "isort>=5.10.1",
+    "mypy",
+    "mypy-zope",
```

Comment:
Note that the reason we don't pin mypy, mypy-zope, and black in the Synapse config here is that we assume developers are using the `poetry.lock` file.

Since we don't use that for sygnal (yet?) we should probably just pin versions here.

Reply:
(I think that makes sense at least, @DMRobertson any thoughts?)

Reply:
Yeah, we want to pin the CI/linter tooling _somewhere_ so that CI doesn't break underneath us.